### PR TITLE
Update message endpoints

### PR DESCRIPTION
**Pull request by paulinebanye**
Created an endpoint to edit a specific message in the collection.
Checks for valid org_id, room_id, and message_id to retrieve message
Ensures that only the sender of the message can edit it.

Pending:
Tests

### CODE REVIEW

**Comment by paulinebanye** [comment]
Closed the old one because I felt there were too many commits.
Took everyone's suggestions into consideration and made adjustments. 

@man-of-mind @zxenonx @AI-fae @o4codes @mykie88

**Comment by o4codes** [comment]
@zxenonx
Kindly look through this PR, as I made some contributions to it.

@AI-fae 
Please go through the PR, I changed the endpoint route of the send message to /orgs/:org_id/rooms/:room_id/messages
Therefore sender_id would be passed in the body of the request. This would enable the same class be used for the update_message endpoint.

**Comment by paulinebanye** [comment]
@o4codes Good job on the edits. Both endpoints work without issues.

**Review by man-of-mind — APPROVED** [review]
Well done bosses @pauline-banye @o4codes

**Review by FlorenceEgwu — APPROVED** [review]
Good work!
 @o4codes and @pauline-banye.

**Review by o4codes — APPROVED** [review]

File: backend/utils/db.py [hunk 1]
```diff
@@ -107,14 +107,14 @@ async def read(
...
-            "options": options,
-        }
+        request_body = dict(
+            collection_name=collection_name,
+            filter=query,
+            object_id=resource_id,
+            organization_id=self.organization_id,
+            plugin_id=self.plugin_id,
+            options=options,
+        )
```
Comment (zxenonx) [inline backend/utils/db.py]: It's generally preferable to use {} notation to declare a dict. This is more "pythonic". 
Moreover, from a performance view, declaring a dictionary with dict() is much slower than with {}.

**Review by zxenonx — APPROVED** [review]
Thanks @pauline-banye and @o4codes.

File: backend/utils/db.py (hunk 1 above)
Comment (o4codes) [reply backend/utils/db.py]: Thanks.
@pauline-banye 
Could you help resolve this ?

Comment (paulinebanye) [reply backend/utils/db.py]: Hi @o4codes, I could look at this but not immediately unfortunately.

Currently out n I won't be home till much later tonight and it definitely won't get sorted before the FE meeting

Comment (zxenonx) [reply backend/utils/db.py]: I've made the change. @pauline-banye @o4codes

**Pull request merged**